src/net.c

```diff
@@ -111,8 +111,12 @@ send_msg(struct http_reply *http, int do_post, char *scep_operation,
 		if (waited_sec >= W_flag)
 			return (1);
 
-		sleep(++fail_count);
-		waited_sec += fail_count;
+		waited_sec += ++fail_count;
+		if (waited_sec <= W_flag) {
+			sleep(fail_count);
+		} else {
+			sleep(W_flag + fail_count - waited_sec);
```

Comment:
I would rather do `++fail_count` only when `waited_sec + fail_count <= W_flag` and here simply wait for `W_flag + fail_count - waited_sec`.
This current implementation changes the logic because `waited_sec` no longer means that the wait has been done.

Reply:
I'm not sure I understood your tips, `waited_sec` is now set to real sleep time.

Reply:
Perfect, thanks!